XbSymbolDatabase.c

```diff
@@ -181,9 +187,47 @@ void XbSymbolSetOutputMessage(xb_output_message_t message_func)
     output_func = message_func;
 }
 
-void XbSymbolOutputMessage(xb_output_message mFlag, const char* message) {
-    if (output_func != 0) {
-        output_func(mFlag, message);
+bool XbSymbolSetOutputVerbosity(xb_output_message verbose_level)
+{
+    if (verbose_level < XB_OUTPUT_MESSAGE_MAX) {
+        output_verbose_level = verbose_level;
+        return true;
+    }
+    // Return false only when requested level is invalid.
+    return false;
+}
+
+void XbSymbolBypassRevLimit(bool bypass_limit)
+{
+    bStrictRevisionLimit = !bypass_limit;
+}
+
+// Intended for send the message as-is without formating.
+// (To avoid corrupted string when perform a bad coding.
+//  Plus certain message need special character without format.)
+void XbSymbolOutputMessage(xb_output_message mLevel, const char* message)
+{
+    // Check if output function is set.
+    // Plus if pass minimum verbose level to output.
+    if (output_func != NULL && mLevel >= output_verbose_level) {
+        output_func(mLevel, message);
+    }
+}
+
+// Intended to format the message with given extra parameters.
+void XbSymbolOutputMessageFormat(xb_output_message mLevel, const char* format, ...)
+{
+    static char bufferTemp[2048];
```

Comment:
This isn't thread-safe.

Thread extension is *cheap*. I'd suggest keeping this on the stack.

Reply:
*Oops*